### PR TITLE
Milestone 2: query and update animals table

### PR DESCRIPTION
**Pull request by dqarias**
in this milestone we:

### Add column
- Add a column species of type string to your animals table. 
image
image
- Insert new data.
image

### Transaction
- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
image

- Inside a transaction:
  - Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
  - Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
  - Commit the transaction.
  - Verify that change was made and persists after commit.
image

- Inside a transaction delete all records in the animals table, then roll back the transaction. After the rollback verify if all records in the animals table still exists. 
image

- Inside a transaction:
  - Delete all animals born after Jan 1st, 2022.
  - Create a savepoint for the transaction.
  - Update all animals' weight to be their weight multiplied by -1.
  - Rollback to the savepoint
  - Update all animals' weights that are negative to be their weight multiplied by -1.
  - Commit transaction
image
image
image

### Queries

- How many animals are there?
image

- How many animals have never tried to escape?
image

- What is the average weight of animals?
image

- Who escapes the most, neutered or not neutered animals?
image

- What is the minimum and maximum weight of each type of animal?
image

- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
image

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -8,3 +9,45 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name NOT IN ('Gabumon');
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <=17.3;
+
+/* Milestone 2*/
+
+-- Transaction 1 
+BEGIN;
+UPDATE animals SET species='unspecified';
+ROLLBACK;
```
Comment (Ismailco) [inline queries.sql]: - [ ] Before `ROLLBACK` kindly consider verifying if the changes are made successfully. you can add `SELECT species from animals;` before the ROLLBACK and after the ROLLBACK

File: queries.sql [hunk 2]
```diff
@@ -8,3 +9,45 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+
+-- Transaction 2
+BEGIN;
+UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species='pokemon' WHERE species IS NULL;
+COMMIT;
```
Comment (Ismailco) [inline queries.sql]: - [ ] Before COMMIT kindly consider verifying if the changes are made successfully. you can add `SELECT species from animals;` before and after the `COMMIT`

File: queries.sql [hunk 3]
```diff
@@ -8,3 +9,45 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+
+-- Transaction 3
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
```
Comment (Ismailco) [inline queries.sql]: - [ ] Same here, please verify your changes you can use this query `SELECT COUNT(*) FROM ANIMALS;` to do that

File: queries.sql [hunk 4]
```diff
@@ -8,3 +9,45 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+-- How many animals have never tried to escape?
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+-- Who escapes the most, neutered or not neutered animals?
+SELECT SUM(escape_attempts), neutered FROM animals GROUP BY neutered;
+-- What is the minimum and maximum weight of each type of animal?
+SELECT MIN(weight_kg), MAX(weight_kg), species from animals GROUP BY species;
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31';
```
Comment (Ismailco) [inline queries.sql]: - [ ] Your result should be groped by animal type, you can add `GROUP BY species` so your query can be like: 
```
 SELECT species, AVG(escape_attempts) FROM animals
   GROUP BY species
   WHERE  date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/12/31';;
```

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @dqarias, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (dqarias) [reply queries.sql]: HI @Ismailco, I did it you can see it in the screenshot that I put in the PR.
image

File: queries.sql (hunk 2 above)
Comment (dqarias) [reply queries.sql]: HI @Ismailco, I did it you can see it in the screenshot that I put in the PR.
image

File: queries.sql (hunk 3 above)
Comment (dqarias) [reply queries.sql]: HI @Ismailco, I did it you can see it in the screenshot that I put in the PR. 
image
It wasn't a requirement to copy on queries.sql. I've put only like reference is the reason that I didn't put the verify sentences on the file.

File: queries.sql (hunk 4 above)
Comment (dqarias) [reply queries.sql]: Thanks, I've just done it

File: queries.sql (hunk 1 above)
Comment (Ismailco) [reply queries.sql]: @dqarias, yes you did. but you should add all the queries in the queries.sql file

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**